### PR TITLE
Update Tokenfactory unit tests

### PR DESCRIPTION
**Pull request by Kbhat1**
- Adds more unit tests to tokenfactory
- Tests message validation
- Tests create, mint, burn
- Tests module account creation

### CODE REVIEW

File: x/tokenfactory/keeper/bankactions_test.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/sei-protocol/sei-chain/x/tokenfactory/types"
+)
+
+func (suite *KeeperTestSuite) TestMultipleMintsPriorToDeferredSettlement() {
```
Comment (philipsu522) [inline x/tokenfactory/keeper/bankactions_test.go:11]: can we also test multiple burns?

Comment (philipsu522) [reply x/tokenfactory/keeper/bankactions_test.go:11]: can we also test multiple mints/burns interlaced with invalid ones (i.e. incorrect denom / admin)?

File: x/tokenfactory/types/denoms_test.go [hunk 2]
```diff
@@ -67,3 +67,65 @@ func TestDecomposeDenoms(t *testing.T) {
 		})
 	}
 }
+
+func TestGetTokenDenom(t *testing.T) {
+	// appparams.SetAddressPrefixes()
```
Comment (philipsu522) [inline x/tokenfactory/types/denoms_test.go]: nit - rm

File: x/tokenfactory/types/denoms_test.go [hunk 3]
```diff
@@ -67,3 +67,65 @@ func TestDecomposeDenoms(t *testing.T) {
...
+	for _, tc := range []struct {
+		desc     string
+		creator  string
+		subdenom string
+		valid    bool
+	}{
+		{
```
Comment (philipsu522) [inline x/tokenfactory/types/denoms_test.go:83]: other edge cases (if we support):
- empty subdenom
- non standard utf-8/ascii

File: x/tokenfactory/keeper/bankactions_test.go (hunk 1 above)
Comment (Kbhat1) [reply x/tokenfactory/keeper/bankactions_test.go:11]: All Added. Please check latest commit with `TestMultipleInterleavedMintsBurns`

File: x/tokenfactory/types/denoms_test.go (hunk 2 above)
Comment (Kbhat1) [reply x/tokenfactory/types/denoms_test.go]: Check latest commit

File: x/tokenfactory/types/denoms_test.go (hunk 3 above)
Comment (Kbhat1) [reply x/tokenfactory/types/denoms_test.go:83]: Added, please check latest commit

**Review by philipsu522 — APPROVED** [review]

**Pull request merged**